packages/core/agent/src/plugins/epoch-monitor/epoch-monitor.ts

```diff
@@ -110,4 +74,49 @@ export class EpochMonitor extends AbstractPlugin {
     this._spaceStates.forEach((state) => state.subscriptions.forEach((subscription) => subscription.unsubscribe()));
     this._spaceStates.clear();
   }
+
+  // TODO(burdon): Subscribe to space members to update address book.
+  private async _monitorSpace(space: Space, state: SpaceState) {
+    state.subscriptions.push(
+      space.members.subscribe((members) =>
+        log('updated', { members: members.map((member) => member.identity.identityKey) }),
+      ),
+    );
+
+    // Monitor spaces owned by this agent.
+    if (this._client!.halo.identity.get()!.identityKey.equals(space.internal.data.creator!)) {
+      log('monitoring', { space: state.spaceKey });
+
+      // Listen for epochs.
+      const limit = this._options.limit ?? DEFAULT_EPOCH_LIMIT;
+      state.subscriptions.push(
+        space.pipeline.subscribe(async (pipeline) => {
+          // TODO(burdon): Rather than total messages, implement inequality in timeframe?
+          assert(checkCredentialType(pipeline.currentEpoch!, 'dxos.halo.credentials.Epoch'));
+          const timeframe = pipeline.currentEpoch?.subject.assertion.timeframe;
+          const currentEpoch = timeframe.totalMessages();
+          const totalMessages = pipeline.currentDataTimeframe?.totalMessages() ?? 0;
+          log('updated', {
+            space: space.key,
+            totalMessages,
+            currentEpoch,
+            epochTriggered: state.epochTriggered,
+          });
+
+          // Guard race condition (epoch triggered while processing pipeline update).
+          if (state.epochTriggered) {
```

Comment:
```suggestion
          if (state.epochTriggered !== undefined) {
```

cause 0 == false in javascript